mersenne-31/src/lib.rs

```diff
@@ -188,6 +188,41 @@ impl Field for Mersenne31 {
         let p29_1 = p28.exp_power_of_2(3) * p1_1;
         Some(p29_1)
     }
+
+    // We hard code computing the 5'th root for rescue.
+    fn exp_u64(&self, power: u64) -> Self {
+        if self.is_zero() {
```

Comment:
nit: assuming this is done as an optimization, I think it might be best to skip just since 0 should be rare in things like arithmetic hashes